dspace/config/modules/authentication-password.cfg

```diff
@@ -26,3 +26,22 @@
 # SHA-256, SHA-384, and SHA-512 should be available, but you may have 
 # installed others.  If not set, SHA-512 will be used.
 # authentication-password.digestAlgorithm = SHA-512
+
+###### Validate Password Robustness Configuration ######
+# (by default is enabled, to disable, either comment out this configuration or set it to an empty value)
+# This regular expression is used to validate password during creation of EPerson
+# or during the patch of password.
+# NOTE: when you configure a custom regex, you have to update the rules on angular's i18n end,
+# variable: "profile.security.form.notifications.error.robust-password".
+#
+# The following regex applies subsequent rules:
+# 1) (?=.*?[a-z])             - the password must contain at least one lowercase character
+# 2) (?=.*?[A-Z])             - the password must contain at least one uppercase character
+# 3) (?=\\S*?[0-9])           - the password must contain at least one numeric character
+# 4) (?=\\S*?[!?$@#$%^&+=])   - the password must contain at least one of the following special character: !?$@#$%^&+=
+# 5) {8\,15}                  - the password must be at least 8 and at most 15 characters long
+# REMARK: {8\,15} - the slash in this regex is an exception of the Apache library, as "," is a special character,
+#                   consequently to interpret it correctly you have to add the slash in front
+authentication-password.regex-validation.pattern = ^(?=.*?[a-z])(?=.*?[A-Z])(?=\\S*?[0-9])(?=\\S*?[!?$@#$%^&+=]).{8\,15}$
```

Comment:
@Micheleboychuk and @corrad82-4s  : Per @artlowel 's suggestion & after reviewing [current NIST Password recommendations ](https://www.auditboard.com/blog/nist-password-guidelines/)(which also say password length is more important than complexity), I'd recommend a compromise approach here. 

1. Leave the current, complex example in our new `authentication-password.cfg` file, but **comment it out**.   This would mean though that we'd need to copy this pattern into the **tests** that depend on it, by using `configurationService.setProperty()`
2. Change the default settings for DSpace to be _at least_ 8 characters as that is the minimum recommendations from NIST (I realize NIST is a USA institution, but I couldn't find a European or other institution with as clear password recommendations.  Additionally, it looks like most corporations like Google, Microsoft, etc have standardized on a 8-character minimum.)
```
# By default, DSpace just requires a password of 8 or more characters.  
# However, we recommend most sites consider either increasing the required length or complexity (see example above)
authentication-password.regex-validation.pattern = ^.{8,}$
```

Does this sound reasonable @artlowel and @Micheleboychuk ?  (Also, we should double check that this new regex is setting is correct...I haven't tested it yet)

(If we'd rather not simplify to 8+ characters, I'm also OK with bumping this to 12+ characters. However, I feel a default of 20 characters seems like it could be too annoying to many users, as no other services appear to require such long passwords at this time.)

Reply:
That sounds reasonable, however I would definitely bump it up to 12, or even 16. 
8 seems insecure no matter how many special characters you require.

Reply:
@artlowel : I feel 16 is too long of a default setting (again, as no other sites require that length).  I'd much rather go with either 8 or 12. 

Using the same [password entropy calculator](https://www.omnicalculator.com/other/password-entropy) you mentioned, if you type in 8 lowercase characters, it still shows "Your password is... strong. Well done!".  That is the minimum it also uses to display this message, as 6 or 7 characters do not show that "strong" setting.

So, this leaves me to believe that 8-characters has become the "bare minimum for a strong password" (as I said, this appears to be the magic number for most every service I've checked).

Reply:
Fair enough